proxy/haproxy.js

```diff
@@ -0,0 +1,84 @@
+const placements = require('./placement');
+
+function buildBucketBackend(bucket, ips) {
+  return `backend ${bucket}_backend
+  mode http
+  http-request set-header X-Forwarded-Host %[req.hdr(Host)]
+${ips}`;
+}
+
+function getServerString(bucket, servers) {
+  let serverString = '';
+  let i = 0;
+  // eslint-disable-next-line
+  for (const server of servers) {
+    serverString += `  server ${bucket}_${i}_server ${server} ssl check verify none\n`;
+    i += 1;
+  }
+  return serverString;
+}
+
+function getBucketBackends() {
+  const { config } = placements;
+  let backends = '';
+  // eslint-disable-next-line
+  for (const bucket of Object.keys(config)) {
+    const servers = getServerString(bucket, config[bucket]);
+    backends += `${buildBucketBackend(bucket, servers)}\n`;
```

Comment:
is servers correct here? should be an ip no?

Reply:
Yes, the `servers` at line 26 is actually something like

`server 0_0_server 167.235.28.60:443 ssl check verify none` where 0_0 is the bucket number and index (since each bucket may have more than 1 fdms) and the ip next to is the actual IP of the FDM